### DISCUSSION TITLE
What is the purpose of PUBLIC_ORIGIN or PUBLIC_HOST?

### DISCUSSION THREAD

**Discussion by trey-sedate** [teableio/teable#856] [Q&A] [answered]
I'm trying to run teable CE locally using docker compose. My configuration includes local redis and local docker minio. My env file is not changed from the docs.

Unsurprisingly, I am getting a 500 error when trying to register as a new user because it is trying to connect to 192.168.x.x (the default value for PUBLIC_HOST in the docs) which doesn't exist.

Clearly I need to modify PUBLIC_ORIGIN and PUBLIC_HOST, as indicated in the docs.

But what should I modify these values to be?

I want my entire setup running locally. My understanding of teable is that I should be able to do this. I don't want any external network connections (incoming or outgoing).

It seems natural to me to use localhost or 127.0.0.1 but this is said not to work in the docs.

What is my misunderstanding?

**Comment by tea-artist** [comment, marked as answer]
Teable's deployment model requires using an external IP or domain name to access MinIO, rather than localhost or 127.0.0.1. This is because Teable needs to access the file service (MinIO) from both the server-side and client-side. The server uses MinIO's presignedUrl API to authorize file access, while the client directly accesses MinIO through signed URLs. In Docker containers, 127.0.0.1 always points to the current container, preventing Teable from correctly accessing MinIO. Although the server could use internal DNS (service name "minio") to access MinIO, this would generate presignedUrls with "minio" as the hostname, making it impossible for the client to access correctly.

The current workaround is to use an external IP or domain name for MinIO access. If you prefer not to expose Teable to external networks, you can opt for Teable's built-in file service instead of enabling MinIO. We're actively exploring better solutions to address this issue.

**Reply by trey-sedate** [reply]
Understood, thank you for explaining. In the meantime, I did find a solution using https://stackoverflow.com/a/61214488 as a guide.

1. Adding a line to my `/etc/hosts` file: `127.0.0.1 my-minio-localhost-alias`
2. Then in my `.env` file, setting `PUBLIC_HOST=my-minio-localhost-alias`
3. Finally, modifying the `teable-storage` service in `docker-compose.yml` to add an alias to the `teable` network. See below for my modifications:

```diff
  teable-storage:
    image: minio/minio:RELEASE.2024-02-17T01-15-57Z
    ports:
      - "9000:9000"
      - "9001:9001"
    environment:
      - MINIO_ACCESS_KEY=${MINIO_ACCESS_KEY}
      - MINIO_SECRET_KEY=${MINIO_SECRET_KEY}
    volumes:
      - teable-storage:/data:rw
    networks:
-     - teable
+     teable:
+       aliases: 
+         - ${PUBLIC_HOST}
    command: server /data --console-address ":9001"
```

Using this, I was able to successfully upload a file.

**Reply by trey-sedate** [reply]
As a brief aside, does this also apply to local configurations using redis? I noticed `PUBLIC_ORIGIN` in the `.env` file of the docs in that section that enables redis.

**Reply by tea-artist** [reply]
> As a brief aside, does this also apply to local configurations using redis? I noticed `PUBLIC_ORIGIN` in the `.env` file of the docs in that section that enables redis.

Redis only needs to be accessed on the backend, so it is not affected
